src/tools/sdktool/adddeviceoperation.cpp

```diff
@@ -414,6 +422,23 @@ bool AddDeviceOperation::exists(const QVariantMap &map, const QString &id)
     return false;
 }
 
+void addVirtualMachineParameters(const Operation::KeyValuePairList &extra, Operation::KeyValuePairList *dev)
+{
+    QString virtualMachineName;
+    for (const auto &value : extra) {
+        if (value.key.contains(QString::fromStdString(Mer::Constants::MER_DEVICE_VIRTUAL_MACHINE))){
+            virtualMachineName = value.value.toString();
+            break;
+        }
+    }
+    if (!virtualMachineName.isEmpty()) {
+        Mer::Internal::MerVirtualBoxManager virtualBoxManager;
+        const auto vmInfo = virtualBoxManager.fetchVirtualMachineInfo(virtualMachineName, true);
+        dev->append(Operation::KeyValuePair(QLatin1String(Mer::Constants::VDI_SIZE_MB), QVariant(vmInfo.vdiSizeOnDisk)));
+        dev->append(Operation::KeyValuePair(QLatin1String(Mer::Constants::VDI_PATH), QVariant(vmInfo.vdiPath)));
```

Comment:
I think these should be passed on command line as well. Do not forget the sdktool is meant to be used by installer, not by human user.